FirebaseSessions/Sources/FirebaseSessions.swift

```diff
@@ -93,10 +94,15 @@ protocol SessionsProvider {
       // On each session start, first update Settings if expired
       self.settings.updateSettings()
       self.identifiers.generateNewSessionID()
-      let event = SessionStartEvent(identifiers: self.identifiers, appInfo: self.appInfo)
-      DispatchQueue.global().async {
-        self.coordinator.attemptLoggingSessionStart(event: event) { result in
+      // Generate a session start event only when session data collection is enabled.
+      if self.settings.sessionsEnabled {
```

Comment:
At some point I want to add tests for this code, maybe in a follow up.

Reply:
Acknowledged. The current test setup does not allow for testing (modular or abstract testing) this easily. So, will differ this on a future change to make such refactors.